operator/.gitignore

```diff
@@ -28,7 +28,7 @@ go.work
 **/test/e2e/abis/**
 
 # Generated by contractpkg tool
-**/config/samples/core_v1alpha1_smartcontractdeployment_*.yaml
+# **/config/samples/core_v1alpha1_smartcontractdeployment_*.yaml
```

Comment:
I'm assuming this might be problematic - but I agree we should be committing codegen files and assessing their diffs if they ever do change.

My experience was there a lot of files being generated/copied around locally, as someone wanting to make a contribution it was a bit convoluted to debug the Helm chart installation process as a result.

Reply:
@dwertent @onelapahead - please pause before we check in 100s of KB of data into our github history, on every recompile of a smart contract we do.

I believe developers will hate a github repo of terrible bloat like that.

Reply:
I removed this code from the Helm chart because Helm was having trouble templating the files. When I adjusted the files to allow Helm to template them successfully, the code failed to decode the objects properly.

In the long term, I think the best solution would be to create separate CRDs for the auto-generated data in the status, or to have the code load the status without creating a complex dependency between the status and the nested object.

For a quick fix to get the system working, I considered simply submitting the files. If we don’t want to go with this approach, another option is to upload the generated files as artifacts, allowing users to pull them from the release. This alternative is less intuitive but avoids committing the generated files and adding them to the Git history.